eng/scripts/TypeSpec-Validation.ps1

```diff
@@ -1,16 +1,45 @@
 [CmdletBinding()]
 param (
   [switch]$CheckAll = $false,
+  [int]$Shard = 0,
+  [int]$TotalShards = 0,
```

Comment:
Why not default to `$TotalShards=1`?  Then we don't need any special-case logic, because 1 shard is just the degenerate case of sharding.